src/couch_mrview/src/couch_mrview.erl

```diff
@@ -480,7 +480,11 @@ map_fold(#full_doc_info{} = FullDocInfo, OffsetReds, Acc) ->
     case couch_doc:to_doc_info(FullDocInfo) of
         #doc_info{id = Id, revs = [#rev_info{deleted = false, rev = Rev} | _]} = DI ->
             Value = {[{rev, couch_doc:rev_to_str(Rev)}]},
-            map_fold({{Id, Id}, Value}, OffsetReds, Acc#mracc{doc_info = DI});
+            NS = couch_util:get_value(namespace, Acc#mracc.args#mrargs.extra),
+            case Id of
+                <<"_design", _/binary>> when NS =:= <<"_non_design">> -> {ok, Acc};
```

Comment:
I think we could use the `-define(DESIGN_DOC_PREFIX, "_design/").` macro in there, and it should already be included from `couch_db.hrl` just to make it a tiny bit neater